bitacross-worker/README.md

```diff
@@ -1 +1,200 @@
+{"jsonrpc":"2.0","result":"0x7d014101d9197274039df1280452819ede02d0867aa57185251d19c9e2c74bd22d1f3b8e1db031b068e3ee7229631a804c1d03e2d9af7055851ac9609dae5e8c7c8dccf4961b31a5cc98bfa356fa262a7376525300a2c03d6f2b59e28578fdcee00000","id":1}
+```
+
+6. Get result from response `result` field. It's a hex representation of scale encoded `RpcReturnValue`.  In case of success, the signature can be obtained from `value` property (it's aes encrypted).
+
+Types definitions:
+
+```rust
+pub type RequestAesKey = [u8; 32];
+
+#[derive(
+Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen, EnumIter, Ord, PartialOrd,
+)]
+pub enum Identity {
+    // web2
+    #[codec(index = 0)]
+    Twitter(IdentityString),
+    #[codec(index = 1)]
+    Discord(IdentityString),
+    #[codec(index = 2)]
+    Github(IdentityString),
+
+    // web3
+    #[codec(index = 3)]
+    Substrate(Address32),
+    #[codec(index = 4)]
+    Evm(Address20),
+    // bitcoin addresses are derived (one-way hash) from the pubkey
+    // by using `Address33` as the Identity handle, it requires that pubkey
+    // is retrievable by the wallet API when verifying the bitcoin account.
+    // e.g. unisat-wallet: https://docs.unisat.io/dev/unisat-developer-service/unisat-wallet#getpublickey
+    #[codec(index = 5)]
+    Bitcoin(Address33),
+}
+
+pub enum LitentryMultiSignature {
+    /// An Ed25519 signature.
+    #[codec(index = 0)]
+    Ed25519(ed25519::Signature),
+    /// An Sr25519 signature.
+    #[codec(index = 1)]
+    Sr25519(sr25519::Signature),
+    /// An ECDSA/SECP256k1 signature.
+    #[codec(index = 2)]
+    Ecdsa(ecdsa::Signature),
+    /// An ECDSA/keccak256 signature. An Ethereum signature. hash message with keccak256
+    #[codec(index = 3)]
+    Ethereum(EthereumSignature),
+    /// Same as above, but the payload bytes are prepended with a readable prefix and `0x`
+    #[codec(index = 4)]
+    EthereumPrettified(EthereumSignature),
+    /// Bitcoin signed message, a hex-encoded string of original &[u8] message, without `0x` prefix
+    #[codec(index = 5)]
+    Bitcoin(BitcoinSignature),
+    /// Same as above, but the payload bytes are prepended with a readable prefix and `0x`
+    #[codec(index = 6)]
+    BitcoinPrettified(BitcoinSignature),
+}
+
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, PartialEq, Eq, Clone, Debug)]
+pub struct EthereumSignature(pub [u8; 65]);
+
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, PartialEq, Eq, Clone, Debug)]
+pub struct BitcoinSignature(pub [u8; 65]);
+
+#[derive(
+Encode, Decode, Copy, Clone, Default, PartialEq, Eq, TypeInfo, MaxEncodedLen, Ord, PartialOrd,
+)]
+pub struct Address20([u8; 20]);
```

Comment:
From Polkadot-js, I can see there is a `AccountId20` type coming from Substrate internals – could our `Address*` be redundant?

Reply:
Yes, but this type comes from `frontier` - not from `Substrate` so it would require extra dependency.